frontend/src/GenerateHeats/UpdateSeedTime.jsx

```diff
@@ -0,0 +1,86 @@
+import { Stack } from "@mui/material";
+import dayjs from "dayjs";
+import { useState } from "react";
+import { useForm } from "react-hook-form";
+import "../App.css";
+import AlertBox from "../components/Common/AlertBox.jsx";
+import { SmmApi } from "../SmmApi.jsx";
+import { formatSeedTime } from "../utils/helperFunctions.js";
+import SeedTimeForm from "./SeedTimeForm.jsx";
+
+const UpdateSeedTime = ({ eventId, athlete, onUpdate, onCancel }) => {
+  const [error, setError] = useState(false);
+  const [submitted, setSubmitted] = useState(false);
+
+  const {
+    handleSubmit,
+    control,
+    formState: { isValid },
+  } = useForm({
+    defaultValues: {
+      seed_time: "",
+      date: dayjs(Date.now()),
+    },
+    mode: "onChange",
+  });
+
+  // AlertBox variables
+  let typeAlert = error ? "error" : "success";
+  let message = error ? error : "Seed time updated successfully!";
+
+  const handleClearError = () => {
+    console.log("Clear error");
+    setError(false);
+    setSubmitted(false);
+  };
+
+  const onSubmit = async (data) => {
+    setSubmitted(true);
```

Comment:
Setting submitted triggers the rendering of the alert box. However, when an error occurs, the success alert box appears first, followed by the error alert box.

Reply:
Moved setSubmitted after API calling